capnp/src/lib.rs

```diff
@@ -169,16 +170,210 @@ pub struct Error {
     pub kind: ErrorKind,
 
     /// Human-readable failure description.
+    #[cfg(feature = "alloc")]
     pub description: String,
 }
 
 /// The general nature of an error. The purpose of this enum is not to describe the error itself,
 /// but rather to describe how the client might want to respond to the error.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum ErrorKind {
```

Comment:
I think that we should add [`#[non_exhaustive]`](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute) on this, so that we can later add new variants as a non-breaking change. (See [`std::io::ErrorKind`](https://doc.rust-lang.org/std/io/enum.ErrorKind.html) for comparison.)